src/ansible_compat/prerun.py

```diff
@@ -0,0 +1,427 @@
+        if not re.match(r"[a-z0-9][a-z0-9_]+\.[a-z][a-z0-9_]+$", fqrn):
+            msg = (
+                """\
+Computed fully qualified role name of %s does not follow current galaxy requirements.
+Please edit meta/main.yml and assure we can correctly determine full role name:
+
+galaxy_info:
+role_name: my_name  # if absent directory name hosting role is used instead
+namespace: my_galaxy_namespace  # if absent, author is used instead
+
+Namespace: https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
+Role: https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names
+
+As an alternative, you can add 'role-name' to either skip_list or warn_list.
+"""
+                % fqrn
+            )
+            if 'role-name' in options.warn_list:
+                _logger.warning(msg)
+            else:
+                _logger.error(msg)
+                sys.exit(INVALID_PREREQUISITES_RC)
+    else:
+        # when 'role-name' is in skip_list, we stick to plain role names
+        if 'role_name' in yaml['galaxy_info']:
+            role_namespace = _get_galaxy_role_ns(yaml['galaxy_info'])
+            role_name = _get_galaxy_role_name(yaml['galaxy_info'])
+            fqrn = f"{role_namespace}{role_name}"
+        else:
+            fqrn = pathlib.Path(".").absolute().name
+    p = pathlib.Path(f"{options.cache_dir}/roles")
+    p.mkdir(parents=True, exist_ok=True)
+    link_path = p / fqrn
+    # despite documentation stating that is_file() reports true for symlinks,
+    # it appears that is_dir() reports true instead, so we rely on exits().
+    target = pathlib.Path(options.project_dir).absolute()
+    if not link_path.exists() or os.readlink(link_path) != str(target):
+        if link_path.exists():
+            link_path.unlink()
+        link_path.symlink_to(target, target_is_directory=True)
+    _logger.info(
+        "Using %s symlink to current repository in order to enable Ansible to find the role using its expected full name.",
+        link_path,
+    )
+
+
+def _prepare_ansible_paths() -> None:
+    """Configure Ansible environment variables."""
+    library_paths: List[str] = []
+    roles_path: List[str] = []
+
+    for path_list, path in (
+        (library_paths, "plugins/modules"),
+        (library_paths, f"{options.cache_dir}/modules"),
+        (collection_list, f"{options.cache_dir}/collections"),
+        (roles_path, "roles"),
+        (roles_path, f"{options.cache_dir}/roles"),
+    ):
+        if path not in path_list and os.path.exists(path):
+            path_list.append(path)
+
+    _update_env('ANSIBLE_LIBRARY', library_paths)
+    _update_env(ansible_collections_path(), collection_list)
+    _update_env('ANSIBLE_ROLES_PATH', roles_path, default=ANSIBLE_DEFAULT_ROLES_PATH)
+
+
+def _update_env(varname: str, value: List[str], default: str = "") -> None:
+    """Update colon based environment variable if needed. by appending."""
+    if value:
```

Comment:
Use a guard expression here.

Reply:
Excuse my ignorance but could you show me how it should look like? (maybe even include a link an article).

Reply:
Guard expressions are simple conditionals that interrupt the control flow early. Like "if X: return" or "if Y: continue/break" (for loops).

Typically, when you have something like:
```python
if True:
    a huge
    indented block
    of code
```

you could dedent that block of code if you just invert that if-conditional expression:
```python
if not True:
    return

a huge
indented block
of code
```

And so this refactoring approach is called "guard expression". It allows you have less highly nested code.